internal/controllers/device_data_controller.go

```diff
@@ -526,6 +528,96 @@ func (d *DeviceDataController) GetVehicleStatus(c *fiber.Ctx) error {
 	return c.JSON(ds)
 }
 
+// GetVehicleStatusV2 godoc
+// @Description Returns the latest status update for the vehicle with a given token id.
+// @Tags        device-data
+// @Param       tokenId path int true "token id"
+// @Produce     json
+// @Success     200 {object} response.Device
+// @Failure     404
+// @Router      /v2/vehicle/{tokenId}/status [get]
+func (d *DeviceDataController) GetVehicleStatusV2(c *fiber.Ctx) error {
+	tis := c.Params("tokenID")
+	claims := c.Locals("tokenClaims").(pr.CustomClaims)
+
+	privileges := claims.PrivilegeIDs
+
+	ti, ok := new(big.Int).SetString(tis, 10)
+	if !ok {
+		return fiber.NewError(fiber.StatusBadRequest, fmt.Sprintf("Couldn't parse token id %q.", tis))
+	}
+
+	//tid := pgtypes.NewNullDecimal(new(decimal.Big).SetBigMantScale(ti, 0))
+	userDeviceNFT, err := d.deviceAPI.GetUserDeviceByTokenID(c.Context(), ti.Int64())
+	if err != nil {
+		d.log.Err(err).Msg("grpc error retrieving NFT metadata.")
+		return err
+	}
+
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return fiber.NewError(fiber.StatusNotFound, "NFT not found.")
+		}
+		d.log.Err(err).Str("token_id", tis).Msg("Database error retrieving NFT metadata or NFT not found")
+		return err
+	}
+
+	if userDeviceNFT == nil {
+		return fiber.NewError(fiber.StatusNotFound, "NFT not found.")
+	}
+
+	deviceData, err := models.UserDeviceData(
+		models.UserDeviceDatumWhere.UserDeviceID.EQ(userDeviceNFT.Id),
+		models.UserDeviceDatumWhere.Signals.IsNotNull(),
+		models.UserDeviceDatumWhere.UpdatedAt.GT(time.Now().Add(-90*24*time.Hour)),
+	).All(c.Context(), d.dbs().Reader)
+	if errors.Is(err, sql.ErrNoRows) || len(deviceData) == 0 {
```

Comment:
This seems iffy. If `err != nil` then my guess is `deviceData` will have length zero. Can we ever hit the below condition?